jgrapht-core/src/test/java/org/jgrapht/alg/flow/PushRelabelMinimumSTCutTest.java

```diff
@@ -52,6 +55,46 @@
         return new PushRelabelMFImpl<>(network);
     }
 
+    public void testSmall()
+    {
+        SimpleDirectedWeightedGraph<Integer, DefaultWeightedEdge> network =
+            new SimpleDirectedWeightedGraph<>(DefaultWeightedEdge.class);
+        network.addVertex(0);
+        network.addVertex(1);
+        network.addVertex(2);
+        network.addVertex(3);
+        network.addVertex(4);
+        network.addVertex(5);
+        network.setEdgeWeight(network.addEdge(2, 4), 62.0);
+        network.setEdgeWeight(network.addEdge(3, 4), 52.0);
+        network.setEdgeWeight(network.addEdge(1, 4), 3.0);
+        network.setEdgeWeight(network.addEdge(0, 1), 58.0);
+        network.setEdgeWeight(network.addEdge(2, 0), 67.0);
+        network.setEdgeWeight(network.addEdge(1, 0), 5.0);
+        network.setEdgeWeight(network.addEdge(4, 0), 11.0);
+        network.setEdgeWeight(network.addEdge(4, 1), 46.0);
+        network.setEdgeWeight(network.addEdge(1, 3), 62.0);
+        network.setEdgeWeight(network.addEdge(4, 3), 27.0);
+
+        MinimumSTCutAlgorithm<Integer, DefaultWeightedEdge> prSolver = this.createSolver(network);
+        double cutWeight = prSolver.calculateMinCut(0, 5);
+        assertEquals(0d, cutWeight);
+    }
+    
+    public void testDisconnected()
+    {
+        SimpleDirectedWeightedGraph<Integer, DefaultWeightedEdge> network =
+            new SimpleDirectedWeightedGraph<>(DefaultWeightedEdge.class);
+        network.addVertex(0);
+        network.addVertex(1);
+        network.addVertex(2);
+        network.setEdgeWeight(network.addEdge(0, 1), 1.0);
```

Comment:
perhaps a bit picky but I think it's cleaner to use the Graphs:
Graphs.addEdge(Graph<V,E> g, V sourceVertex, V targetVertex, double weight);
method.